### PR TITLE
Deprecate `{{partial}}`

### PR DESCRIPTION
**Pull request by GavinJoyce**
Part of https://github.com/emberjs/rfc-tracking/issues/38
Supersedes https://github.com/emberjs/ember.js/pull/14972

TODO:

 * [x] Add deprecation warning
 * [x] Add svelting for deprecated code
 * [x] Update docs to mark `{{partial}}` as `@deprecated`
 * [x] Deprecation guide PR: https://github.com/ember-learn/deprecation-app/pull/456

### CODE REVIEW

**Review by rwjblue — COMMENTED** [review]
Do you think there is a way to make it so all of `packages/@ember/-internals/views/lib/system/lookup_partial.js` can be stripped (aka svelted)? Can we inline that module into here:

https://github.com/emberjs/ember.js/blob/ee8a894980fad6664a26ba78c8d949891abf811b/packages/%40ember/-internals/glimmer/lib/resolver.ts

And remove the various reexports and whatnot?

**Comment by GavinJoyce** [comment]
Good call, I've moved `lookup_partial.js` into `resolver.ts` and wrapped it in for svelting

File: packages/@ember/-internals/glimmer/lib/resolver.ts [hunk 1]
```diff
@@ -176,6 +178,62 @@ function lookupComponent(owner: Owner, name: string, options: LookupOptions): Op
...
+
+  templateFor = function(owner: any, underscored: string, name: string) {
+    if (PARTIALS) {
+      if (!name) {
+        return;
+      }
+      assert(`templateNames are not allowed to contain periods: ${name}`, name.indexOf('.') === -1);
+
+      if (!owner) {
+        throw new EmberError(
```
Comment (rwjblue) [inline packages/@ember/-internals/glimmer/lib/resolver.ts]: This error is a bit "wild", but 🤷‍♂

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**